web/src/stream_topic_history.ts

```diff
@@ -46,21 +67,49 @@ export function all_topics_in_cache(sub: StreamSubscription): boolean {
         return true;
     }
```

Comment:
Reading this, I do feel like we should be checking `fetched_stream_ids` and returning `true` if so around here.